dace/codegen/targets/cuda.py

```diff
@@ -955,6 +983,65 @@ def generate_devicelevel_state(self, sdfg, state, function_stream,
                     callsite_stream.write('__syncthreads();', sdfg, state_id)
                     break
 
+        # In GPU_Persistent scopes, states need global barriers between them,
+        # the DFGs inside of a state are independent, so they don't need
+        # synchronization. DFGs in a GPU_Persistent scope are per se executed
+        # by a single thread only. (Device) Maps however can be distributed
+        # across multiple threads
+        elif self._toplevel_schedule == dtypes.ScheduleType.GPU_Persistent:
+
+            # reset streams in GPU persistent maps if the lifetime is scope,
+            # otherwise streams do not behave as expected becasue they are
+            # allocated on host side
+            # TODO discuss how exactly streams should behave (between states)
```

Comment:
Still need to discuss?

Reply:
I don't think so, it was from when we discussed whether to reset the streams each state or not.

Reply:
Yep